src/components/oscal-utils/OSCALCatalogUtils.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * Determines if a control is withdrawn.
+ *
+ * TODO: This is probably 800-53 specific and it should be made more
+ * generic to allow the library to work with other frameworks.
+ * https://github.com/EasyDynamics/oscal-react-library/issues/504
+ *
+ * @param {object} control
+ * @returns a boolean describing whether the control is withdrawn
+ */
+export default function isWithdrawn(control) {
+  return control?.props?.find(
+    (prop) => prop.name === "status" && prop.value === "withdrawn"
+  );
+}
```

Comment:
Let's actually do namespace validation. Some of this code probably belongs in a different file but I am thinking of an API a bit like this (ignore the types if the resulting file has to be JS):


```ts
/**
 * When a namespace is not specified for a property, the assumption is that the
 * namespace is default namespace.
 */
const NIST_DEFAULT_NAMESPACE = "https://csrc.nist.gov/ns/oscal";

export interface PropFilter {
  /**
   * The namespace of the property.
   * 
   * @default https://csrc.nist.gov/ns/oscal
   */
  ns?: string;

  /**
   * The name of the prop to find.
   */
  name: string;

  /**
   * The acceptable value of the property.
   * 
   * Either this property or {@link filter} (but not both) must be specified.
   */
  value?: string;

  /**
   * A function to validate the property value.
   * 
   * Either this property or {@link value} (but not both) must be specified.
   */
  filter?: (value: string) => boolean;
}

/**
 * Return the given namespace, or if undefined, the default namespace.
 */
export function namespaceOf(ns: string | undefined): string {
  return ns ?? NIST_DEFAULT_NAMESPACE;
}

/**
 * Returns the first matching property in the given list.
 * 
 * @param props the list of properties to check
 * @param filter the attributes to match against
 */
export function matchingProp(props: Prop[], filter: PropFilter): Prop | undefined {
  const ns = namespaceof(filter.ns);
  return props.find((prop) =>
    namespaceOf(prop.ns) === namespaceOf(filter.namespace)
    && prop.name === filter.name
    && (!filter.value || prop.value === filter.value)
  );
}

export function isWithdrawn(control: Control): boolean {
  // By default controls are *not* withdrawn
  if (!control.props) {
    return false;
  }
  return !!matchingProp(control.props, { name: "status", value: "withdrawn });
}
```

Besides `isWithdrawn`, most of this probably deserves to exist more in an `oscal-utils/OSCALPropUtils.js`.

Reply:
(Eventually, we may want to support `class` in that filter interface but I do not want to deal with that today)

Reply:
I've gone ahead and added namespace validation for JS - types have been removed as well as the interface for now. There were a couple other variable corrections I had to make, but I was able to figure them out.

Reply:
I've separated `isWithdrawn` from the rest of the functions `oscal-utils/OSCALPropUtils.js` but followed the same logic as mentioned. I've just kept the `isWithdrawn` function in the Catalog Utils class I created. Let me know if something else works better here.